commerce/domain/promotion/cart/calculator/classes/PromotionCalculatorSample.cls

```diff
@@ -0,0 +1,98 @@
+        cartextension.Cart cart = request.getCart();
+        resetAllAdjustments(cart);
+        applyAdjustments(cart);
+    }
+
+    /**
+     * @description Remove cart & cart-item level adjustments, cart validation outputs.
+     * @param cart Holds details about cart
+     */
+    public static void resetAllAdjustments(cartextension.Cart cart) {
+
+        // Remove all cart-level adjustments
+        Iterator<CartExtension.CartAdjustmentGroup> cagIter = cart.getCartAdjustmentGroups().iterator();
+        List<CartExtension.CartAdjustmentGroup> cagToRemove= new List<cartextension.CartAdjustmentGroup>();
+        while(cagIter.hasNext()) {
+            cagToRemove.add(cagIter.next());
+        }
+        // Remove CartItemAdjustments
+        for(CartExtension.CartAdjustmentGroup cag : cagToRemove) {
+            cart.getCartAdjustmentGroups().remove(cag);
+        }
+
+        // Remove all cart-item level adjustments
+        Iterator<CartExtension.CartItem> ciIter = cart.getCartItems().iterator();
+        while(ciIter.hasNext()) {
+
+            // For every cart item, cursor through adjustments
+            CartExtension.CartItem ci = ciIter.next();
+            Iterator<CartExtension.CartItemPriceAdjustment> ciaIter = ci.getCartItemPriceAdjustments().iterator();
+            List<CartExtension.CartItemPriceAdjustment> ciaToRemove= new List<CartExtension.CartItemPriceAdjustment>();
+
+            // For every adjustment, look for a matching external adjustment
+            while(ciaIter.hasNext()) {
+                ciaToRemove.add(ciaIter.next());
+            }
+            for(CartExtension.CartItemPriceAdjustment cia : ciaToRemove) {
+                ci.getCartItemPriceAdjustments().remove(cia);
+            }
+        }
+
+        // Remove promotion cart validation output
+        Iterator<CartExtension.CartValidationOutput> cvoIter = cart.getCartValidationOutputs().iterator();
+        List<CartExtension.CartValidationOutput> cvoToRemove = new List<CartExtension.CartValidationOutput>();
+        while(cvoIter.hasNext()) {
+            cvoToRemove.add(cvoIter.next());
+        }
+        for(CartExtension.CartValidationOutput cvo : cvoToRemove) {
+            if (CartExtension.CartValidationOutputTypeEnum.PROMOTIONS == cvo.getType()) {
+                cart.getCartValidationOutputs().remove(cvo);
+            }
+        }
+    }
+
+
+    /**
+     * @description Apply flat 5 percent discount across all cart items
+     * @param cart Holds details about cart
+     */
+    public static void applyAdjustments(CartExtension.Cart cart) {
+
+        Decimal pctDiscount = -5;
+        Iterator<CartExtension.CartItem> ciIter = cart.getCartItems().iterator();
+        while(ciIter.hasNext()) {
+
+            CartExtension.CartItem ci = ciIter.next();
+            Decimal promotionAdjustment = (ci.getSalesPrice() * (pctDiscount/100) * ci.getQuantity());
```

Comment:
@patricia-bagzai-sf This line has an assumption that SalesPrice is always available on the CartItem, which means Pricing Calculator must be invoked before this code is executed - and that is **not** guaranteed. Would you mind handling case when there is no SalesPrice on the CartItem?